File: trunk/src/test/java/net/esper/regression/pattern/TestEveryOperator.java
package net.esper.regression.pattern;

import junit.framework.TestCase;
import net.esper.regression.support.CaseList;
import net.esper.regression.support.EventCollection;
import net.esper.regression.support.EventCollectionFactory;
import net.esper.regression.support.EventExpressionCase;
import net.esper.regression.support.PatternTestHarness;
import net.esper.support.bean.SupportBeanConstants;

public class TestEveryOperator extends TestCase implements SupportBeanConstants
{
    public void testOp() throws Exception
    {
        EventCollection events = EventCollectionFactory.getEventSetOne(0, 1000);
        CaseList testCaseList = new CaseList();
        EventExpressionCase testCase = null;

        testCase = new EventExpressionCase("every b=" + EVENT_B_CLASS);
        testCase.add("B1", "b", events.getEvent("B1"));
        testCase.add("B2", "b", events.getEvent("B2"));
        testCase.add("B3", "b", events.getEvent("B3"));
        testCaseList.addTest(testCase);

        testCase = new EventExpressionCase("b=" + EVENT_B_CLASS);
        testCase.add("B1", "b", events.getEvent("B1"));
        testCaseList.addTest(testCase);

        testCase = new EventExpressionCase("every (every (every b=" + EVENT_B_CLASS + "))");
        testCase.add("B1", "b", events.getEvent("B1"));
        for (int i = 0; i < 3; i++)
        {
            testCase.add("B2", "b", events.getEvent("B2"));
        }
        for (int i = 0; i < 9; i++)
        {
            testCase.add("B3", "b", events.getEvent("B3"));
        }
        testCaseList.addTest(testCase);

        testCase = new EventExpressionCase("every (every b=" + EVENT_B_CLASS + "())");
        testCase.add("B1", "b", events.getEvent("B1"));
        testCase.add("B2", "b", events.getEvent("B2"));
        testCase.add("B2", "b", events.getEvent("B2"));
        for (int i = 0; i < 4; i++)
        {
            testCase.add("B3", "b", events.getEvent("B3"));
        }
        testCaseList.addTest(testCase);

        testCase = new EventExpressionCase("every( every (every (every b=" + EVENT_B_CLASS + "())))");
        testCase.add("B1", "b", events.getEvent("B1"));
        for (int i = 0; i < 4; i++)
        {
            testCase.add("B2", "b", events.getEvent("B2"));
        }
        for (int i = 0; i < 16; i++)
        {
            testCase.add("B3", "b", events.getEvent("B3"));
        }
        testCaseList.addTest(testCase);

        PatternTestHarness util = new PatternTestHarness(events, testCaseList);
        util.runTest();
    }
}

File: NEsper.Tests/regression/pattern/TestEveryOperator.cs
using System;

using net.esper.regression.support;
using net.esper.support.bean;

using NUnit.Core;
using NUnit.Framework;

namespace net.esper.regression.pattern
{

	[TestFixture]
	public class TestEveryOperator : SupportBeanConstants
	{
		[Test]
		public virtual void  testOp()
		{
			EventCollection events = EventCollectionFactory.GetEventSetOne(0, 1000);
			CaseList testCaseList = new CaseList();
			EventExpressionCase testCase = null;

			testCase = new EventExpressionCase("every b=" + net.esper.support.bean.SupportBeanConstants.EVENT_B_CLASS);
			testCase.Add("B1", "b", events.GetEvent("B1"));
			testCase.Add("B2", "b", events.GetEvent("B2"));
			testCase.Add("B3", "b", events.GetEvent("B3"));
			testCaseList.AddTest(testCase);

			testCase = new EventExpressionCase("b=" + net.esper.support.bean.SupportBeanConstants.EVENT_B_CLASS);
			testCase.Add("B1", "b", events.GetEvent("B1"));
			testCaseList.AddTest(testCase);

			testCase = new EventExpressionCase("every (every (every b=" + net.esper.support.bean.SupportBeanConstants.EVENT_B_CLASS + "))");
			testCase.Add("B1", "b", events.GetEvent("B1"));
			for (int i = 0; i < 3; i++)
			{
				testCase.Add("B2", "b", events.GetEvent("B2"));
			}
			for (int i = 0; i < 9; i++)
			{
				testCase.Add("B3", "b", events.GetEvent("B3"));
			}
			testCaseList.AddTest(testCase);

			testCase = new EventExpressionCase("every (every b=" + net.esper.support.bean.SupportBeanConstants.EVENT_B_CLASS + "())");
			testCase.Add("B1", "b", events.GetEvent("B1"));
			testCase.Add("B2", "b", events.GetEvent("B2"));
			testCase.Add("B2", "b", events.GetEvent("B2"));
			for (int i = 0; i < 4; i++)
			{
				testCase.Add("B3", "b", events.GetEvent("B3"));
			}
			testCaseList.AddTest(testCase);

			testCase = new EventExpressionCase("every( every (every (every b=" + net.esper.support.bean.SupportBeanConstants.EVENT_B_CLASS + "())))");
			testCase.Add("B1", "b", events.GetEvent("B1"));
			for (int i = 0; i < 4; i++)
			{
				testCase.Add("B2", "b", events.GetEvent("B2"));
			}
			for (int i = 0; i < 16; i++)
			{
				testCase.Add("B3", "b", events.GetEvent("B3"));
			}
			testCaseList.AddTest(testCase);

			PatternTestHarness util = new PatternTestHarness(events, testCaseList);
			util.RunTest();
		}
	}
}
